js/util/extractProps.ts

```diff
@@ -0,0 +1,47 @@
+const pick = require('lodash.pick');
+const flatten = require('flat');
+
+export interface Component {
+  //TODO: use the real RN ones
+  props: {
+    [propertyKey: string]: any;
+  };
+}
+
+export interface PropExtractorCriteria {
+  include: string[];
+  exclude: string[];
+}
+
+export interface PropExtractorConfig {
+  [componentName: string]: PropExtractorCriteria;
+}
+
+export const extractProps = (
+  elementName: string,
+  component: Component,
+  config: PropExtractorConfig
+): string => {
+  if (!component || !config[elementName]) {
+    return '';
+  }
+
+  const filteredProps = pick(component.props, config[elementName].include);
+  const flattenedProps = flatten(filteredProps);
+  let propsString = '';
+
+  // Only include props that are primitives.
+  Object.keys(flattenedProps).forEach(key => {
+    if (
+      flattenedProps[key] !== null &&
```

Comment:
I guess this is because of the prettier config, but in the case of binary operations split over multiple lines, I generally prefer the operator to be on the new line, e.g.
```
if (flattendProps[key] !== null
    && flattendProps[key] !== undefined)
```

I'll try to see if we have any guidance on this sort of thing in any internal style guides, but I don't think we do :/

Reply:
Yeah, once we live in the prettier world, you either gotta be able to express it in a prettier config (which may be possible - not sure) or just get used to reading it :-)